File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/DataPropertyPathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.posts.pages.PostPage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class DataPropertyPathType extends PermalinkPathType {

    @Inject
    public DataPropertyPathType() {
        super(1);
    }

    @Override
    public boolean acceptsKey(PostPage post, String key) {
        return post.getData().has(key);
    }

    @Override
    public String format(PostPage post, String key) {
        return post.getData().get(key).toString();
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathTypes/DataPropertyPathType.kt
package com.eden.orchid.posts.permalink.pathTypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.posts.permalink.PermalinkPathType

import javax.inject.Inject

class DataPropertyPathType @Inject
constructor() : PermalinkPathType(1) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return page.data.has(key)
    }

    override fun format(page: OrchidPage, key: String): String? {
        return page.data.get(key).toString()
    }

}

